Kwf/Media.php

```diff
@@ -219,4 +218,44 @@ public static function createCacheId($class, $id, $type)
     {
         return str_replace(array('.', '>'), array('___', '____'), $class) . '_' . str_replace('-', '__', $id) . '_' . $type;
     }
+
+    private static function _generateCacheFilePath($class, $id, $type)
+    {
+        $groupingFolder = substr(md5($id), 0, 3);
+        return Kwf_Config::getValue('mediaCacheDir').'/'.$class.'/'.$groupingFolder.'/'.$id.'/'.$type;
+    }
+
+    public static function collectGarbage($debug)
+    {
+        $cacheFolder = Kwf_Config::getValue('mediaCacheDir');
+        $mediaClasses = array_slice(scandir($cacheFolder), 2);
+        foreach ($mediaClasses as $mediaClass) {
+            if (in_array($mediaClass, array('.gitignore'))) continue;
+            if (is_file($cacheFolder.'/'.$mediaClass)) continue;
+
+            if (!is_instance_of($mediaClass, 'Kwf_Media_Output_ClearCacheInterface')) continue;
+
+            $classFolder = $cacheFolder.'/'.$mediaClass;
+            $groups = array_slice(scandir($classFolder), 2);
```

Comment:
this is not very nice, every time you scan_dir you slice . and ..
eventually glob is better?